engines/alg/game.cpp

```diff
@@ -0,0 +1,595 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "audio/audiostream.h"
+#include "audio/decoders/raw.h"
+#include "common/events.h"
+#include "common/substream.h"
+#include "common/timer.h"
+#include "graphics/cursorman.h"
+#include "graphics/paletteman.h"
+
+#include "alg/graphics.h"
+#include "alg/scene.h"
+
+#include "alg/game.h"
+
+namespace Alg {
+
+Game::Game(AlgEngine *vm) {
+	_vm = vm;
+}
+
+Game::~Game() {
+	_libFile.close();
+	_libFileEntries.clear();
+	delete _rnd;
+	delete[] _palette;
+	delete _videoDecoder;
+	delete _sceneInfo;
+	if (_background) {
+		_background->free();
+		delete _background;
+	}
+	if (_screen) {
+		_screen->free();
+		delete _screen;
+	}
+	for (auto item : *_gun) {
+		if (item) {
+			item->free();
+			delete item;
+		}
+	}
+	for (auto item : *_numbers) {
+		if (item) {
+			item->free();
+			delete item;
+		}
+	}
+	if (_saveSound != nullptr) {
+		delete _saveSound;
+	}
+	if (_loadSound != nullptr) {
+		delete _loadSound;
+	}
+	if (_easySound != nullptr) {
+		delete _easySound;
+	}
+	if (_avgSound != nullptr) {
+		delete _avgSound;
+	}
+	if (_hardSound != nullptr) {
+		delete _hardSound;
+	}
+	if (_skullSound != nullptr) {
+		delete _skullSound;
+	}
+	if (_shotSound != nullptr) {
+		delete _shotSound;
+	}
+	if (_emptySound != nullptr) {
+		delete _emptySound;
+	}
```

Comment:
It's safe to call `delete` on null pointers. Check [here](https://www.reddit.com/r/cpp_questions/comments/no536z/what_happens_if_i_use_delete_on_nullptr/).

Thus, the code above could be simplified as follows:
```suggestion
	delete _saveSound;
	delete _loadSound;
	delete _easySound;
	delete _avgSound;
	delete _hardSound;
	delete _skullSound;
	delete _shotSound;
	delete _emptySound;
```

Reply:
fixed